phac-nml/irida-next
# Group bot accounts UI

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI for group bot accounts (listing, removing, adding).

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/212ba3d3-b862-4dd5-acb7-70e6a95b95b1)

![image](https://github.com/phac-nml/irida-next/assets/25466211/efd25983-8545-4c36-8eca-5223e62a8ece)

![image](https://github.com/phac-nml/irida-next/assets/25466211/abc49f9a-7d26-43bb-b24a-8f2c648a8a8c)

![image](https://github.com/phac-nml/irida-next/assets/25466211/dc405ca9-c3fb-46ca-963d-2ecd230add03)

![image](https://github.com/phac-nml/irida-next/assets/25466211/5bc9c320-51bb-4dca-84c2-02480a6ed9bc)

![image](https://github.com/phac-nml/irida-next/assets/25466211/640f601d-d4b2-4772-8332-31830d930b91)

![image](https://github.com/phac-nml/irida-next/assets/25466211/44ea287d-0c64-4799-a313-d25baf58cb39)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com` or any other user
2. Create a group
3. From the sidebar on the project show page, click `Settings` and then `Bot Accounts`
4. You should not see any bot accounts
5. Click the `New bot account` button
6. Fill in the token name, access level, option expiration date for token, and select at least 1 scope
7. Click `Submit`
8. You should see a table with the bot account that was created and above the table a section which lists the personal access token
9. Go to the `Members` page for the group
10. You should see the bot account that was created listed as a `Bot`
11. Go back to the `Bot Accounts` page for the group
12. Click the ellipsis for the bot account and click `Remove`. Confirm that you would like to remove the bot account
13. The table should update and no bot accounts should be listed
14. Go back to the `Members` page for the group
15. Verify that the bot account has also been removed from the group members

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
